app/gui/src/main/scala/org/bitcoins/gui/dlc/DLCPane.scala

```diff
@@ -136,17 +135,145 @@ class DLCPane(glassPane: VBox) {
     spacing = 10
   }
 
+  private val tableView: TableView[DLCStatus] = {
+    val paramHashCol = new TableColumn[DLCStatus, String] {
+      text = "Temp Contract Id"
+      prefWidth = 150
+      cellValueFactory = { status =>
+        new StringProperty(status,
+                           "Temp Contract Id",
+                           status.value.tempContractId.hex)
+      }
+    }
+
+    val contractIdCol = new TableColumn[DLCStatus, String] {
+      text = "Contract Id"
+      prefWidth = 150
+      cellValueFactory = { status =>
+        val contractIdStr = status.value match {
+          case _: DLCStatus.Offered | _: DLCStatus.Accepted =>
+            ""
+          case signed: SignedDLCStatus =>
+            signed.contractId.toHex
+        }
+
+        new StringProperty(status, "Contract Id", contractIdStr)
+      }
+    }
+
+    val statusCol = new TableColumn[DLCStatus, String] {
+      text = "Status"
+      prefWidth = 150
+      cellValueFactory = { status =>
+        new StringProperty(status, "Status", status.value.statusString)
+      }
+    }
+
+    val initiatorCol = new TableColumn[DLCStatus, String] {
+      text = "Initiator"
+      prefWidth = 80
+      cellValueFactory = { status =>
+        val str = if (status.value.isInitiator) {
+          "Yes"
+        } else {
+          "No"
+        }
+        new StringProperty(status, "Initiator", str)
+      }
+    }
+
+    val collateralCol = new TableColumn[DLCStatus, String] {
+      text = "Collateral"
+      prefWidth = 110
+      cellValueFactory = { status =>
+        val num = if (status.value.isInitiator) {
+          status.value.offer.totalCollateral.toLong
+        } else {
+          status.value
+            .asInstanceOf[AcceptedDLCStatus]
```

Comment:
a match may have been a little nicer :P

Reply:
The thing is that if `isInitiator` is false, it will always be at least a `AcceptedDLCStatus`

Reply:
Used a match